### PR TITLE
Api own

### PR DESCRIPTION
**Pull request by alexgul4enko**

### CODE REVIEW

File: src/app/common/session/authMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          let eKey = key
+          if(nonFieldErrorKeys.includes(key)) {
+            eKey = '_error'
+          }
+          return {
+            ...res,
+            [eKey]: Array.isArray(value) ? value[0] : value,
+          }
+        }, {})
+        throw new SubmissionError(errors)
```
Comment (legendar) [inline src/app/common/session/authMiddleware.js]: please move this to `withReduxForm` HOC (in separate PR )

File: src/app/common/session/authMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      }
+      if(get(response, 'status') === 401) {
+        store.dispatch(logout())
+      }
```
Comment (legendar) [inline src/app/common/session/authMiddleware.js]: code duplication detected =)

File: src/app/init.js [hunk 3]
```diff
@@ -1,16 +1,18 @@
 import 'polyfills' // should be first
 import '../styles/index.scss'
-import { createStore, applyMiddleware, combineReducers, compose as reduxCompose } from 'redux'
-import { reducer as form } from 'redux-form'
-import { createEpicMiddleware, combineEpics } from 'redux-observable'
+import API from './api'
+import { resourcesReducer } from 'ds-resource'
+import { cacheMiddleware, persistReducer } from 'ds-cache'
+import { thunkMiddleware, composeReducers, combineReducers } from 'ds-thunk'
```
Comment (legendar) [inline src/app/init.js]: please rename thunkMiddleware. e.g. `promisableActionMiddleware` (in separate PR)

File: webpack.config.babel.js [hunk 4]
```diff
@@ -96,6 +97,9 @@ module.exports = createConfig([
       contentBase: path.resolve(`${process.env.OUTPUT_PATH}`),
       port: process.env.DEV_SERVER_PORT || 3000,
       overlay: true,
+      watchOptions: {
+        poll: true,
+      },
```
Comment (legendar) [inline webpack.config.babel.js]: please remove this option. it will cause performance issues

File: src/app/init.js (hunk 3 above)
Comment (alexgul4enko) [reply src/app/init.js]: https://github.com/django-stars/ds-thunk/pull/1

https://github.com/django-stars/frontend-skeleton/pull/205/commits/d442a62f8515aad2f7223742e4c96d95eee570f9

also i think it is better for now to rename ds-thunk as ds-redux-helpers and we will need to rename repo as well. I will update yarn lock later

File: package.json [hunk 5]
```diff
@@ -34,9 +34,15 @@
   },
   "dependencies": {
     "@sentry/browser": "^5.6.1",
+    "abortcontroller-polyfill": "^1.3.0",
     "bootstrap": "^4.0.0",
     "core-decorators": "^0.20.0",
+    "ds-api": "https://github.com/django-stars/ds-frontend-api",
+    "ds-cache": "https://github.com/django-stars/ds-fe-cache",
+    "ds-resource": "https://github.com/django-stars/frontend-skeleton-contrib",
```
Comment (legendar) [inline package.json]: please update url

File: package.json [hunk 6]
```diff
@@ -34,9 +34,15 @@
...
+    "ds-thunk": "https://github.com/django-stars/ds-thunk",
     "lodash": "^4.17.5",
+    "memoize-one": "^5.1.1",
```
Comment (legendar) [inline package.json]: extra dependency

File: src/app/api.js [hunk 7]
```diff
@@ -1,179 +1,13 @@
...
-  options() {
-    return this.request('OPTIONS')
-  }
-
-  delete() {
-    return this.request('DELETE')
-  }
+export default api
+export {
+  QS,
```
Comment (legendar) [inline src/app/api.js]:
```
export const QS = new QueryParams()
```

File: src/app/common/session/authMiddleware.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+
+export function logout() {
+  return function(dispatch) {
+    dispatch({
+      type: LOGOUT_ACTION,
+    })
+    dispatch(reset())
+  }
+}
```
Comment (legendar) [inline src/app/common/session/authMiddleware.js]: please move the action to session

**Pull request merged**